### PR TITLE
feat(mppi): implement auto-lambda tuning (fixes #13)

### LINKED ISSUE #13: feat(mppi): implement lambda (temperature) auto-tuning
Implement an automatic tuning mechanism for the lambda (temperature) parameter in MPPI controllers. 

Currently, lambda is a fixed parameter. Auto-tuning (e.g., based on target Effective Sample Size or entropy) can improve robustness across different cost landscapes and noise levels.

Objectives:
- Add parameters for target ESS and adaptation rate.
- Implement the lambda update law in the MPPI node.
- Apply it to the MPPI Acc controller (and potentially TQ).
- Ensure smooth transitions and stability.

### PR DESCRIPTION
**Pull request by riccardo-enr**
This PR implements automatic lambda (temperature) tuning for the MPPI controller based on Effective Sample Size (ESS).

### Changes:
- **Core Logic**: Added  in  to adjust lambda dynamically.
- **Node Update**:  now calculates ESS and updates lambda if  is enabled.
- **Observability**: Added  and  topics for real-time monitoring.
- **Config**: Updated  with tuning parameters (, , etc.).
- **Compatibility**:  explicitly disables auto-tuning for now to preserve existing behavior.

Fixes #13

### CODE REVIEW

**Comment by riccardo-enr** [comment]
## Testing Update: Low ESS, Lambda Hitting Max

### Current Behavior
- **ESS**: Low (well below target of 250)
- **Lambda**: Hitting lambda_max (currently 1.0)
- **Expected lambda**: Papers suggest 1e-2 to 1e-3 for trajectory tracking

### Analysis: Cost Scaling Issue
The auto-tuner wants lambda > 1.0, but typical values are ~0.001-0.01. This suggests **costs are too large relative to lambda**.

With current gains (Q_pos=500, Q_vel=50) and H=20:
- Position error 1m → 500 cost/timestep
- Over 20 timesteps → ~10,000 cost

Weight computation: `exp(-(cost - min_cost) / lambda)`
- If cost_range ≈ 1000 and lambda = 0.001: `exp(-1000/0.001)` = `exp(-1M)` ≈ 0
- Result: Only best sample has weight → ESS ≈ 1

### Possible Solutions
1. **Reduce Q/R gains** - Scale down position/velocity costs
2. **Increase lambda_max** - Allow larger lambda (but contradicts literature)
3. **Add cost normalization** - Normalize by H or timestep
4. **Check sigma** - May need more noise diversity

### Next Steps
- Add logging to see actual min/max cost values
- Consider cost normalization or gain rescaling

**Comment by riccardo-enr** [comment]
## Implementation Update: Jang'\''s Gradient-Based Method

Implemented Minhyuk Jang'\''s gradient-based lambda auto-tuning method, but still encountering issues.

### What Was Implemented

**1. Cost Normalization by H**
- Normalized costs by horizon length (average cost per timestep)
- Reduced cost scale by factor of H (~20)

**2. Jang'\''s Gradient-Based Method** ()
- Objective: J(λ) = α·Fluctuation(λ) + β·Performance(λ)
- Fluctuation: Variance of weighted mean (low λ → high jitter)
- Performance: Expected cost (high λ → includes bad trajectories)
- Uses finite differences for gradient estimation
- Gradient descent: λ ← λ - η·∇J(λ)

**3. New Parameters**
```yaml
auto_lambda: true
use_gradient_method: true
alpha_fluctuation: 1.0
beta_performance: 1.0
learning_rate: 0.1
lambda_perturbation: 0.01
lambda_min: 0.0001
lambda_max: 100.0
```

### Current Status: Still Not Working

Even with both cost normalization and the gradient-based method, the controller is not performing well. Possible remaining issues:

1. **Gain scaling** - Q_pos=500, Q_vel=50 may still be fundamentally wrong scale
2. **Cost normalization insufficient** - May need to normalize by expected cost, not just H
3. **Lambda initialization** - Starting at 0.001 may be too small
4. **Implementation bugs** - The finite-difference gradient or objective computation may have issues

### Next Steps to Try

1. Add debug logging to see actual cost values and lambda evolution
2. Try normalizing by expected cost magnitude (e.g., Q_pos * position_error²)
3. Test with much lower gains (Q_pos: 500 → 5)
4. Initialize lambda at higher value (e.g., 1.0)
5. Consider analytical gradient instead of finite differences

### Files Modified

- `src/i_mppi/mppi_control/include/mppi_control/mppi_common.hpp`
- `src/i_mppi/mppi_control/src/mppi_acc_node.cpp`
- `src/fuel_planner/plan_bringup/config/mppi_acc.yaml`

Will continue debugging tomorrow.
